src/App.js

```diff
@@ -26,19 +26,26 @@ function App() {
   }
 
   function handleRemoveItem(id) {
-    setShoppingCart(
-      shoppingCart.map((item) => {
-        // TODO: Remove Item from Cart if Quantity is 0
-
-        // if (item.id === id && item.quantity === 1) {
-        //   setShoppingCart(shoppingCart.filter((item) => item.id !== id));
-        // }
-        if (item.id === id) {
-          return { ...item, quantity: item.quantity - 1 };
-        }
-        return item;
-      })
-    );
+    shoppingCart.map((item) => {
+      if (item.id === id && item.quantity === 1) {
```

Comment:
Why do you check item.quantity for 1 and not for 0?

Reply:
if you have 1x item and you click "-" you have 0x of that item. And so it is directly deleted.